File: ToyLanguageInterpretor/src/utils/ISymbolTable.java
package utils;
import java.util.List;
import java.util.Map;

public interface ISymbolTable<K,V> {
	void add(K key,V value);
	boolean contains(K key);
	V getValue(K key);
	void setValue(K key,V value);
	Iterable<Map.Entry<K, V>> getAll();
	ISymbolTable<K,V> clone();
	List<V> values();
}

File: C#/ToyLanguageC/utils/ISymbolTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToyLanguageC.utils
{
    interface ISymbolTable<K,V>
    {
        void add(K key, V value);
        bool contains(K key);
        V getValue(K key);
        List<KeyValuePair<K, V>> getAll();
    }
}
